lib/player/player.js

```diff
@@ -736,6 +741,36 @@ shaka.player.Player.prototype.setPlaybackRate = function(rate) {
 
 
 /**
+ * @param {Object} restrictions An Object containing the restrictions
+ *     on video tracks that you wish to set. For example, passing
+ *     {'maxBandwidth': 700000, 'minBandwidth': 200000} will restrict
+ *     playback to video tracks with bandwidth between 700000 and 200000.
+ * @export
+ */
+shaka.player.Player.prototype.setRestrictions = function(restrictions) {
+  for (var key in restrictions) {
+    if (this.restrictions_.hasOwnProperty(key)) {
+      this.restrictions_[key] = restrictions[key];
+    }
+  }
+  this.videoSource_.setRestrictions(this.restrictions_);
+};
+
+
+/**
+ * @return {Object} restrictions Contains the current restrictions
+ * @export
+ */
+shaka.player.Player.prototype.getRestrictions = function() {
```

Comment:
I like this idea. But just with respect to the issue below of replacing struct on Restrictions, the way I would like to implement the clone() method would be similar to this because rather than explicitly setting each property because then there is no need to modify this method if new restrictions are added at some point. This will prohibit me from replacing struct, even if the changes below are made. Am I right about this?